### DISCUSSION TITLE
Mac training var reference

### DISCUSSION THREAD

**Discussion by xlinx** [bmaltais/kohya_ss#1185] [General]
### Running on 
```

- v21.8.2 OK
- v21.8 FAIL 
- v21.7.16 OK

```

### Setup
```accelerate config```
This machine/NO/NO/NO
### RUN and Var
`accelerate launch --num_cpu_threads_per_process=10 "./train_network.py" --enable_bucket --pretrained_model_name_or_path="~/_ML/stable-diffusion-webui/models/Stable-diffusion/v1-5-pruned -emaonly.safetensors" --train_data_dir="~/_ML/_TrainingX/modelName/img" --resolution="512,512"--output_dir="~/_ML/_TrainingX/modelName/out" --logging_dir="~/_ML/_TrainingX/modelName/log"--network_alpha="2" --save_model_as=safetensors --network_module=networks.lora --text_encoder_lr=5e-05 --unet_lr=0.0001 --network_dim=40 --output_name="viwa_x" --lr_scheduler_num_cycles="10" --no_half_vae --learning_rate="0.0001" --lr_scheduler="cosine" --lr_warmup_steps="440" --train_batch_size="1"--max_train_steps="4400" --save_every_n_epochs="1" --mixed_precision="no" --save_precision="float"--seed="1234" --cache_latents --cache_latents_to_disk --optimizer_type="AdamW"--max_data_loader_n_workers="0" --bucket_reso_steps=64 --bucket_no_upscale
`
### key setup
```
--mixed_precision="no"
--save_precision="float"
--optimizer_type="AdamW"
--no_half_vae
```

### Time
Mac M2 32G using full GPU start training w/11 photos on 4400 steps around **1hr10min**.
For anyone using Mac reference.
thx ありがとう Kohya ( If one day come to our country, please allow us to take you to enjoy our delicious Taiwanese street food.)

**Comment by itortouch** [comment]
This is the only way it worked for me. Thanks!

- v.21.8.9 working on mi M2 MacBook Pro

**Comment by mitonj** [comment]
Same here. The only way I managed to run TI on macOS with M2 CPU

**Comment by Akossimon** [comment]
also only now got it running , after 3 weeks of no results. 


this is my environment where it runs now, Apple M2 Pro mit 12‑Core CPU, 19‑Core GPU, 16‑Core Neural Engine and 32GB ram
am doing LORA training of 270 images and it will go for around 60 hours. so 5 days.

i wonder if there are better faster MAc settings in Kohya  that one could do 

below a screenshot pdf (searchable pdf) of all my settings which now worked, so for newbies with these screenshot  it is much easier to take over the settings 

[Kohya SS Mac mimi M2pro 32gb LORA training Settings ABBY OCR.pdf](https://github.com/user-attachments/files/15741535/Kohya.SS.Mac.mimi.M2pro.32gb.LORA.training.Settings.ABBY.OCR.pdf)


[Kohya SS Mac mimi M2pro 32gb LORA training Settings ABBY OCR.pdf]
